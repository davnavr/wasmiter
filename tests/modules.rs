#[test]
fn basic_module() {
    let wat = r#"(module
    (func (export "add_five") (param i32) (result i32)
        local.get 0
        i32.const 5
        i32.add))
"#;
    let wasm = wat::parse_str(wat).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn exception_handling() {
    let wasm = wat::parse_str(include_str!("modules/exception_handling.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn all_the_things() {
    let wasm = wat::parse_str(include_str!("modules/all_the_things.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}
