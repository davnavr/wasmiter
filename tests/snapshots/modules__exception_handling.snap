---
source: tests/modules.rs
expression: "wasmiter::parse_module_sections(wasm.as_slice()).unwrap().display_module()"
---
(type (; 0 ;) (func (param i32) (result i32)))
(type (; 1 ;) (func (param i32) (result)))

(import "env" "DoTheThings" (func (; 0 ;) (type 0)))

;; function section count = 1
(tag (type 1))

(func (type 0) ;; code size = 21
  try (result i32)
    local.get 0
    i32.const 0x0000FFFF (; 65535 signed, 65535 unsigned ;)
    i32.add
    call 0
  catch 0
    drop
    i32.const 0xFFFF0000 (; -65536 signed, 4294901760 unsigned ;)
    end
)

(; UNRECOGNIZED (0) @ 0x4D to 0x63
[
    046E616D_65011001__000D646F_5F746865,
    5F746869_6E6773,
]
;)

